bot/cogs/free.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from datetime import datetime
+
+from discord import Colour, Embed, Member, utils
+from discord.ext.commands import BucketType, Context, command, cooldown
+
+
+log = logging.getLogger(__name__)
+
+
+class Free:
+    """Tries to figure out which help channels are free."""
+
+    PYTHON_HELP_ID = 356013061213126657
+    TIME_INACTIVE = 300
+
+    @command(name="free", aliases=('f',))
+    @cooldown(1, 60.0, BucketType.channel)
+    async def free(self, ctx: Context, user: Member = None, seek: int = 2):
+        """
+        Lists free help channels by likeliness of availability.
+        :param user: accepts user mention, ID, etc.
+        :param seek: How far back to check the last active message.
+
+        seek is used only when this command is invoked in a help channel.
+
+        When seek is 2, we are avoiding considering the last active message
+        in a channel to be the one that invoked this command.
+
+        When seek is 3 or more, a user has been mentioned on the assumption
+        that they asked if the channel is free or they asked their question
+        in an active channel, and we want the message before that happened.
+        """
+        free_channels = []
+        python_help = utils.get(ctx.guild.categories, id=self.PYTHON_HELP_ID)
+
+        if user is not None and seek == 2:
+            seek = 3
+        elif not 0 < seek < 10:
+            seek = 3
+
+        for channel in python_help.channels:
+            if channel.id == ctx.channel.id:
+                messages = await channel.history(limit=seek).flatten()
+                msg = messages[seek-1]
+            else:
+                messages = await channel.history(limit=1).flatten()
+                msg = messages[0]
+
+            inactive = (datetime.utcnow() - msg.created_at).seconds
+            if inactive > self.TIME_INACTIVE:
+                free_channels.append((inactive, channel.id))
+
+        embed = Embed()
+        embed.colour = Colour.gold()
+        embed.title = "**Looking for a free help channel?**"
+
+        if user is not None:
+            embed.description = f"**Hey <@{user.id}>!**\n\n"
```

Comment:
If we do the following, we can let discord.py handle the mention instead of manually parsing it:
```suggestion
            embed.description = f"**Hey {user.mention}!**\n\n"
```

Reply:
This is awesome, I did not know about this.